### PR TITLE
feat: add GA workflows & tox as test runner

### LINKED ISSUE #6: Github Workflow management
Si vous avez besoin de droit spécifique sur git pour ce sujet, demandez moi :) 

- créer un hook pour éviter qu'on puisse push sur master
- créer des tags de priorisation (0 : urgent , 1 : medium, 2: long terme)
- demander à l'équipe si ils aimeraient avoir d'autres tags à implémenter
- ajouter un groupe de personne (ie notre équipe tech) qui serait automatiquement dans les reviewer des PR
- ajouter un système d'alerting (slack ou email) dès qu'une PR est en "in review" (peut être via gitAction ? ou tout simplement à voir dans les settings github)

### LINKED ISSUE #17: Remplacer flake8 + isort par Ruff
@benoit-cty a [suggéré](https://github.com/dataforgoodfr/Bloom/pull/11#issuecomment-1441371772) que nous remplacions `flake8` + `isort` par [Ruff](https://github.com/charliermarsh/ruff-pre-commit) dans nos pre-commit hooks.

### LINKED ISSUE #19: Add tests + CI for each PR
=> quels tests ? simplement lancer le job et voir qu'on a bien X lignes (X correspondant au nombre de bateau)

=> quel framework ? pytest ?

### PR DESCRIPTION
**Pull request by SaboniAmine**
fix #19
fix #17 
fix #6

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -12,13 +18,13 @@
 logger = getLogger()
 
 
-def scrap_vessels_with_marine_traffic():
+def scrap_vessels_with_marine_traffic() -> None:
```
Comment (Erawpalassalg) [inline app.py:21]: Note sure why there are type annotations here? This looks like something which should be done in the context of #14

File: src/domain/vessel.py [hunk 2]
```diff
@@ -24,33 +23,31 @@ def to_list(self):
         ]
 
     @validator("timestamp", pre=True)
-    def format_timestamp(cls, value):
+    def format_timestamp(self, value: str) -> datetime:
```
Comment (Erawpalassalg) [inline src/domain/vessel.py]: Quoting from pydantic's docs:

> validators are "class methods", so the first argument value they receive is the UserModel class, not an instance of UserModel

Not sure why these were changed?

**Review by Erawpalassalg — COMMENTED** [review]
Overall things looks fine, but I can see several issues.

- Commits are scrambled.
   - Some looks useless (e.g. 064b294)
   - Some have parts belonging to other commits (e.g. fc8d72f fixes `print` from 52340d)

This feels like an issue for two linked reasons:
  - It ties each commit to several issues, and so makes their name irrelevant
  - It makes the commits virtually irreversible
  
- This PR seems to also try to partly fix #14, which, to me, hasn't yet been discussed and probably shouldn't be addressed right away
- This PR only addresses only part of #19: adding the CI, but doesn't add tests as discussed there. This is not a problem per se, but #19 should be updated and a new issue for the test be made if we want to merge this PR as is

File: pyproject.toml [hunk 3]
```diff
@@ -1,7 +1,11 @@
+[build-system]
+requires = ["poetry-core"]
+build-backend = "poetry.core.masonry.api"
+
 [tool.poetry]
-name = "bloom"
+name = "src"
```
[the remark this answers is not in the document]
Comment (SaboniAmine) [reply pyproject.toml:6]: You are right, I'll change the main folder name to `bloom` then.
In order to install package & run tests, the package name should be the same as root folder name.

File: pyproject.toml [hunk 4]
```diff
@@ -11,12 +15,109 @@ apscheduler = "3.9.1"
...
+    "TRY",
+    "UP",
+    "W",
+    "YTT",
+]
+unfixable = []
+
+exclude = [
+    "bloom_scraper.py",
+    "bloom_scraper_crontab.py",
```
[the remark this answers is not in the document]
Comment (SaboniAmine) [reply pyproject.toml:97]: They are only excluded from linting, as they do not respect some of rules introduced with Ruff (commented code, type annotations, trailing commas...). Anyway I guess that they are not useful anymore, as they have been rewritten through the app.
I would propose to remove them.

File: app.py (hunk 1 above)
Comment (SaboniAmine) [reply app.py:21]: Those return types have been introduced to respect linting rules.
Let's solve the #14 issue in another PR.

**Comment by SaboniAmine** [comment]
Thanks for the review!
Commits have been rewritten to have a more coherent history.
I've also removed `bloom_scraper.py` & `bloom_scraper_crontab.py` scripts as discussed above (not useful anymore).
OK for you @RonanMorgan ?

**Comment by RonanMorgan** [comment]
yes ok for me :)

**Review by Erawpalassalg — APPROVED** [review]
It still feels like the major part of 345439391c0bdb2188a545e53465ba14c27d11bf belong to 52340d0b4e9a58e1806888ad07bec9bfd3f1eda0

**Pull request merged**